### PR TITLE
Introduce Usage docs

### PR DESCRIPTION
**Pull request by kburger**
🚧 Documentation on how to use reference implementation specific components, focused on the resource definitions, shapes, and the API.

### CODE REVIEW

**Comment by kburger** [comment]
@rajaram5 if you have any input on the API usage docs, let me know.

File: docs/usage/api-usage.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+``POST`` ``/${prefix}``         :ref:`create-metadata`
+``PUT``  ``/${prefix}/${uuid}`` :ref:`update-metadata`
+======== ====================== ========================
+
+.. _retrieve-metadata:
+
+Retrieving metadata
+-------------------
+
+Retrieving metadata is open for GET requests without authorization.
```
Comment (rajaram5) [inline docs/usage/api-usage.rst:55]: Just a minor thing, can you indicate that ur using dataset resources as an example here and the prefix for that resource is `dataset` ?

File: docs/usage/usage.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Resource definitions can be :ref:`created <create-resource-definition>`, :ref:`modified <modify-resource-definition>`, or :ref:`deleted <delete-resource-definition>`.
+
+.. _create-resource-definition:
+
+Creating resource definitions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Creating a new resource definition.
+
+**Name** defines the human readable name for a resource definition. This name is used in the admin dashboard to identify the definitions, and does not affect a definition on the functional level.
```
Comment (rajaram5) [inline docs/usage/usage.rst:26]: can you give an example here?

File: docs/usage/usage.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+**URL Prefix** defines the URL path for a resource. This context path should be a unique identifier, unique in the scope of the other resource defined within the FAIR Data Point instance.
```
Comment (rajaram5) [inline docs/usage/usage.rst:28]: can you give an example here?

File: docs/usage/usage.rst [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+**Target Class URIs** links the :ref:`shape definitions <shapes>` to a resource definition. In the current implementation, each shape that should be applied to the resource must be listed here. The expected URI value must match the ``sh:targetClass`` value of the shape definition. A common example is to list the ``dcat:Resource`` shape target class along with the specific subclass for the resource, like ``dcat:Dataset``.
+
+**Children** defines child resources, if any. This applies when the resource acts as a parent resource for other types of resources. Children are defined by the following properties to provide directives for both the server and the client side.
+
+* *Child Resource* links to the child's *resource definition*. The child resource must be defined beforehand.
+* *Child Relation URI* defines the predicate IRI that links the parent to the child on the metadata instance level.
```
Comment (rajaram5) [inline docs/usage/usage.rst:35]: can you give an example here?

**Review by rajaram5 — COMMENTED** [review]
I added some minor comments

**Review by rajaram5 — APPROVED** [review]
Looks good to me. Also the SHACL part

**Review by MarekSuchanek — APPROVED** [review]
Looks good to me 👍

**Comment by kburger** [comment]
Any comments or additions before merging this initial description of the usage docs?

**Review by janslifka — APPROVED** [review]

**Comment by MarekSuchanek** [comment]
> Any comments or additions before merging this initial description of the usage docs?

I think its great, we can continually improve in the future 👍 

Currently there is mentioned version `1.6.0` but it will get to the "latest" with `1.8.0` where we will change version tags also on other places.

**Pull request merged**